maro/simulator/scenarios/oncall_routing/business_engine.py

```diff
@@ -578,7 +579,11 @@ def _on_action_received(self, event: CascadeEvent) -> None:
         action_by_route: Dict[str, List[Action]] = defaultdict(list)
         for action in actions:
             assert isinstance(action, Action)
-            action_by_route[action.route_name].append(action)
+            if isinstance(action, PostponeAction):
+                self._postponed_order_ids.add(action.order_id)
+            else:
+                assert isinstance(action, AllocateAction)
+                action_by_route[action.route_name].append(action)
```

Comment:
Shall we add a protection here, in case the agent processes an order that has been postponed previously?
Logic could be:
```python
if action.order_id in self._postponed_order_ids:
    self._postponed_order_ids.remove(action.order_id)
```

Reply:
great idea!

Reply:
Add one more validation check